client/src/main/java/io/pravega/client/stream/impl/IdempotentEventStreamWriterImpl.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.stream.impl;
+
+import com.google.common.base.Preconditions;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.segment.impl.SegmentOutputStream;
+import io.pravega.client.segment.impl.SegmentOutputStreamFactory;
+import io.pravega.client.segment.impl.SegmentSealedException;
+import io.pravega.client.stream.AckFuture;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.IdempotentEventStreamWriter;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.Stream;
+import io.pravega.common.Exceptions;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.concurrent.GuardedBy;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class takes in events, finds out which segment they belong to and then calls write on the appropriate segment.
+ * It deals with segments that are sealed by re-sending the unacked events to the new correct segment.
+ * 
+ * @param <Type> The type of event that is sent
+ */
+@Slf4j
+@ToString(of = { "stream", "closed" })
+public class IdempotentEventStreamWriterImpl<Type> implements IdempotentEventStreamWriter<Type> {
+
+    private final Object lock = new Object();
+    private final Stream stream;
+    private final Serializer<Type> serializer;
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+    private final EventWriterConfig config;
+    @GuardedBy("lock")
+    private final SegmentSelector selector;
+    @GuardedBy("lock")
+    private long lastSequence = Long.MIN_VALUE;
+
+    IdempotentEventStreamWriterImpl(Stream stream, UUID writerId, Controller controller,
```

Comment:
Say an application wants to resume from the same writer id. It might be useful to know what the last sequence number successfully written was. Is there a way of obtaining it?